frontend/server/k8s-helper.ts

```diff
@@ -62,6 +63,62 @@ function getNameOfViewerResource(logdir: string): string {
   return 'viewer-' + crypto.SHA1(logdir);
 }
 
+function parseTensorboardLogDir(logdir: string, podTemplateSpec: object): string {
```

Comment:
nit: this is a very complex function, can you add more unit tests for the function itself?

Reply:
Ok, more testing on the way.